tests/Resources/util.test.js

```diff
@@ -0,0 +1,1551 @@
+				eventListeners.push(listener);
+				this.listeners[eventName] = eventListeners;
+			};
+
+			BaseClass.prototype.emit = function (eventName, data) {
+				const eventListeners = this.listeners[eventName] || [];
+				for (const listener of eventListeners) {
+					listener.call(this, data);
+				}
+			};
+
+			function MyStream() {
+				BaseClass.call(this);
+			}
+
+			util.inherits(MyStream, BaseClass);
+
+			MyStream.prototype.write = function (data) {
+				this.emit('data', data);
+			};
+
+			const stream = new MyStream();
+
+			should(stream instanceof BaseClass).eql(true);
+			should(MyStream.super_).eql(BaseClass);
+
+			stream.on('data', data => {
+				data.should.eql('It works!');
+				finished();
+			});
+			stream.write('It works!'); // Received data: "It works!"
+		});
+
+		it('throws TypeError if super constructor is null', () => {
+			function BaseClass() {
+			}
+
+			function MyStream() {
+				BaseClass.call(this);
+			}
+
+			should.throws(() => util.inherits(MyStream, null),
+				TypeError
+			);
+		});
+
+		it('throws TypeError if constructor is null', () => {
+			function BaseClass() {
+			}
+
+			should.throws(() => util.inherits(null, BaseClass),
+				TypeError
+			);
+		});
+
+		it('throws TypeError if super constructor has no prototype', () => {
+			const BaseClass = Object.create(null, {});
+
+			function MyStream() {
+				BaseClass.call(this);
+			}
+
+			should.throws(() => util.inherits(MyStream, BaseClass),
+				TypeError
+			);
+		});
+	});
+
+	describe('#isArray', () => {
+		it('should return true only if the given object is an Array', () => {
+			should.strictEqual(util.isArray([]), true);
+			should.strictEqual(util.isArray(Array()), true);
+			should.strictEqual(util.isArray(new Array()), true);
+			should.strictEqual(util.isArray(new Array(5)), true);
+			should.strictEqual(util.isArray(new Array('with', 'some', 'entries')), true);
+			should.strictEqual(util.isArray({}), false);
+			should.strictEqual(util.isArray({ push: function () {} }), false);
+			should.strictEqual(util.isArray(/regexp/), false);
+			should.strictEqual(util.isArray(new Error()), false);
+			should.strictEqual(util.isArray(Object.create(Array.prototype)), false);
+		});
+	});
+
+	describe('#isRegExp', () => {
+		it('should return true only if the given object is a RegExp', () => {
+			should.strictEqual(util.isRegExp(/regexp/), true);
+			should.strictEqual(util.isRegExp(RegExp(), 'foo'), true);
+			should.strictEqual(util.isRegExp(new RegExp()), true);
+			should.strictEqual(util.isRegExp({}), false);
+			should.strictEqual(util.isRegExp([]), false);
+			should.strictEqual(util.isRegExp(new Date()), false);
+			should.strictEqual(util.isRegExp(Object.create(RegExp.prototype)), false);
+		});
+	});
+
+	describe('#isDate', () => {
+		it('should return true only if the given object is a Date', () => {
+			should.strictEqual(util.isDate(new Date()), true);
+			should.strictEqual(util.isDate(new Date(0), 'foo'), true);
```

Comment:
<!--
  0 failure: 
  1 warning:  tests/Resources/u...
  
  
  DangerID: danger-id-Danger;
  File: tests/Resources/util.test.js;
  Line: 867;
-->

- :warning: tests/Resources/util.test.js line 867 – 'util.isDate' was deprecated since v4.0.0. (node/no-deprecated-api)